navbar.ts

```diff
@@ -0,0 +1,433 @@
+      if (!this.change) {
+        this.navbar.style.display = "none";
+      }
+
+      this.menuElement.style.display = "block";
+      this.change = true;
+      this.anchorElements.forEach((anchor) => {
+        anchor.style.display = "block";
+      });
+    } else {
+      if (this.change) {
+        this.handleClose();
+        this.change = false;
+      }
+      this.menuElement.style.display = "none";
+      this.navbar.style.display = "block";
+    }
+  }
+
+  getAnchorElement(elementName: string): HTMLAnchorElement | null {
+    const anchorElement = Array.from(this.anchorElements).find(
+      (anchor) =>
+        this.parseAttributes(anchor.getAttribute("attr") || "").nm ===
+        elementName
+    );
+    return anchorElement || null;
+  }
+
+  handleMenu(event: Event) {
+    const clickedAnchor = event.currentTarget as HTMLAnchorElement;
+    clickedAnchor.style.display = "none";
+    if (this.mode == "overlay") this.menu.style.display = "block";
+    else if (this.mode == "saccordion") this.menu.style.display = "block";
+    else this.menu.style.display = "block";
+  }
+  handleClose() {
+    this.menu.style.display = "none";
+    this.menuElement.style.display = "block";
+    const len = this.currentPath.length;
+    if (this.activeClickOutsideListener) {
+      document.removeEventListener("click", this.activeClickOutsideListener);
+    }
+    if (len) {
+      const anchor = this.currentPath[len - 1];
+      const attributes = this.parseAttributes(
+        anchor.getAttribute("attr") || ""
+      );
+      this.overlayContainer[attributes.nm][0].style.display = "none";
+    }
+    this.currentPath = [];
+  }
+  handleBack() {
+    const currentAnchor = this.currentPath[this.currentPath.length - 1];
+    const currentAnchorName = this.parseAttributes(
+      currentAnchor.getAttribute("attr") || ""
+    ).nm;
+    this.overlayContainer[currentAnchorName][0].style.display = "none";
+    this.currentPath.pop();
+
+    const cAnchor = this.currentPath[this.currentPath.length - 1];
+    if (cAnchor) {
+      const cAnchorName = this.parseAttributes(
+        cAnchor.getAttribute("attr") || ""
+      ).nm;
+      this.overlayContainer[cAnchorName][0].style.display = "block";
+    }
+  }
+
+  getNavbarLength(): number {
+    let len = 0;
+    const original = this.querySelectorAll(
+      this.attr ? `[${this.attr}]` : "a"
+    ) as NodeListOf<HTMLAnchorElement>;
+    const lengthOfInput = original.length;
+
+    let i = 0;
+    while (i < lengthOfInput) {
+      const anchor = original[i];
+      const attrValue = anchor.getAttribute(this.attr) || "";
+      const attributes = this.parseAttributes(attrValue);
+      const attrName = attributes.nm;
+
+      if (!attributes.pnt) {
+        this.noParentAnchors.push({ elementName: attrName, children: [] });
+        if (!(attrName === "menu"))
+        {
+          len+=anchor.offsetWidth;
+        }
+      }
+      i++;
+    }
+    console.log(len);
+
+    return len+35;

```

Comment:
+35 is aribitrary, we need proper justification for a number to be used. You can either name the constant in a variable or use a formula with const variables so that the readability is good.